pkg/api/v1alpha1/cluster.go

```diff
@@ -582,9 +608,22 @@ func validateGitOps(clusterConfig *Cluster) error {
 	gitOpsRefKind := gitOpsRef.Kind
 	fluxConfigActive := features.IsActive(features.GenericGitProviderSupport())
 
-	if gitOpsRefKind == FluxConfigKind && !fluxConfigActive {
-		return fmt.Errorf("FluxConfig and the generic git provider are not currently supported; " +
-			"to use this experimental feature, please set the environment variable GENERIC_GIT_PROVIDER_SUPPORT to true")
+	if gitOpsRefKind == FluxConfigKind {
```

Comment:
here too, validateGitOps is to validate cluster obj's gitops config, not the fluxconfig itself